File: TouchControl/src/VolumeSlider.java
import java.io.*;

import org.opencv.core.*;

public class VolumeSlider extends Slider
{
	private int volume;

	//Constructors
	public VolumeSlider(Rect sliderDim)
	{
		super(sliderDim);
	}

	public VolumeSlider(Rect sliderDim, Scalar color)
	{
		super(sliderDim, color);
	}

	//Setters

	//Getters

	//Class methods
	@Override
	public void performAction()
	{
		if (validateAction())
			if (volume >= 0 && volume <= numOfDivisions)
			{
				int newVolume = numOfDivisions - division;

				if (volume != newVolume)
					setVolume(newVolume);
			}
	}

	private void setVolume(int volume)
	{
		//Nircmd allows volume changing
		File nircmdPath = new File("./nircmd.exe");

		System.out.println("Volume set to: " + volume);

		//Calculate volume in nircmd terms
		double endVolume = 655.35 * volume;

		try
		{
			//Set volume
			this.volume = volume;
			Runtime.getRuntime().exec(nircmdPath.getCanonicalPath() + " setsysvolume " + endVolume);
		}
		catch (IOException e)
		{
			e.printStackTrace();
		}
	}

	@Override
	public String toString()
	{
		return super.toString();
	}
}

File: src/main/kotlin/me/jakejmattson/touchcontrol/touchables/VolumeSlider.kt
package me.jakejmattson.touchcontrol.touchables

import org.opencv.core.*
import java.io.File

/**
 * Demo Slider - Example application of abstract slider.
 * Controls system volume.
 *
 * @author JakeJMattson
 */
class VolumeSlider(dimensions: Rect, color: Scalar) : Slider(dimensions, color) {
    /**
     * Internal representation of current system volume
     */
    private var previousVolume: Int = 0

    override fun performAction() {
        if (hasDetection()) {
            val newVolume = (numOfDivisions - currentDivision) * (100 / numOfDivisions)

            if (previousVolume != newVolume)
                setVolume(newVolume)
        }
    }

    private fun setVolume(volume: Int) {
        setSystemVolume(volume)
        previousVolume = volume
    }

    private fun setSystemVolume(volume: Int) {
        val nircmd = File("nircmd/nircmd.exe")

        if (!nircmd.exists())
            return

        //Convert volume to nircmd standard
        val nircmdVolume = 655.35 * volume

        //Build command to set system volume
        val command = "${nircmd.absolutePath} setsysvolume $nircmdVolume"

        Runtime.getRuntime().exec(command)
    }
}